rules/no-useless-promise-resolve-reject.js

```diff
@@ -46,6 +46,26 @@ function getFunctionNode(node) {
 	};
 }
 
+function isPromiseCallback(node) {
+	if (
+		node.parent.type === 'CallExpression'
+		&& node.parent.callee.type === 'MemberExpression'
+		&& !node.parent.callee.computed
+		&& node.parent.callee.property.type === 'Identifier'
+	) {
+		const {callee: {property}, arguments: arguments_} = node.parent;
+		if (property.name === 'then') {
+			return arguments_[0] === node || (arguments_[0].type !== 'SpreadElement' && arguments_[1] === node);
+		}
+
+		if (property.name === 'catch' || property.name === 'finally') {
+			return arguments_[0] === node;
```

Comment:
I don’t know how common non-Promise then/catch/finally methods are in real code. Do you think it would make sense to do a length/type check on the arguments list to confirm it’s likely to be a valid Promise call?

Reply:
https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1666/commits/21c2773c6e187c9445bac9d7e2c14826119d022d